math/src/main/java/com/powsybl/math/graph/UndirectedGraphImpl.java

```diff
@@ -572,21 +573,53 @@ public boolean traverse(int[] startingVertices, TraversalType traversalType, Tra
         return true;
     }
 
+    /**
+     * This method is used to get the size of a path considering only the elements validating the predicate
+     * @param path Path to size
+     * @param sizingPredicate Predicate used to filter the elements of the path
+     * @return size of the filtered path
+     */
+    private int filteredSize(TIntArrayList path, Predicate<? super E> sizingPredicate) {
+        return Math.toIntExact(Arrays.stream(path.toArray())
+            .filter(idx -> sizingPredicate.test(edges.get(idx).getObject()))
+            .count());
+    }
+
     /**
      * {@inheritDoc}.
-     *
+     * <p>
      * This method allocates a {@link List} of {@link TIntArrayList} to store the paths, a {@link BitSet} to store the encountered vertices
      * and calls {@link #findAllPaths(int, Function, Function, TIntArrayList, BitSet, List)}.
+     * </p>
+     * In the output, the paths are sorted by size considering the number of switches in each path.
      */
     @Override
     public List<TIntArrayList> findAllPaths(int from, Function<V, Boolean> pathComplete, Function<E, Boolean> pathCancelled) {
+        return findAllPaths(from, pathComplete, pathCancelled, obj -> true);
+    }
+
+    /**
+     * {@inheritDoc}.
```

Comment:
You don't inherit anything here! You didn't put this method in the API. I think it should be added.